### PR TITLE
Implement index out of range checks

### PR DESCRIPTION
**Pull request by quajak**

### CODE REVIEW

File: source/Cosmos.IL2CPU/IL/Ldlen.cs [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+                    : base(aAsmblr)
+                {
+                }
 
-        public override void Execute(_MethodInfo aMethod, ILOpCode aOpCode)
-        {
-            DoNullReferenceCheck(Assembler, DebugEnabled, 4);
+                public override void Execute(_MethodInfo aMethod, ILOpCode aOpCode)
+                {
+                    Assemble(Assembler, DebugEnabled);
```
Comment (charlesbetros) [inline source/Cosmos.IL2CPU/IL/Ldlen.cs]: What are the changes here intended to do? No reason to add an extra method call.

File: source/Cosmos.IL2CPU/IL/Ldelem_Ref.cs [hunk 2]
```diff
@@ -25,7 +25,29 @@ public static void Assemble(Assembler aAssembler, uint aElementSize, bool isSign
     {
       //  stack     = index
       //  stack + 2 = array
-      DoNullReferenceCheck(aAssembler, debugEnabled, 8);
+      // null reference check is done in ldlen
```
Comment (charlesbetros) [inline source/Cosmos.IL2CPU/IL/Ldelem_Ref.cs]: The pattern we use for loading references/pointers is to do the null check at the beginning.

**Comment by quajak** [comment]
For tests see https://github.com/CosmosOS/Cosmos/pull/1165

File: source/Cosmos.IL2CPU/IL/Ldelem_Ref.cs (hunk 2 above)
Comment (quajak) [reply source/Cosmos.IL2CPU/IL/Ldelem_Ref.cs]: Ok changed it

**Pull request merged**